dbms/src/Functions/LeastGreatest.h

```diff
@@ -120,5 +122,289 @@ class FunctionVectorizedLeastGreatest : public IFunction
+        for (size_t i = 0; i < size; ++i)
+        {
+            LeastStringImpl::process(collator, a_data, a_offsets, b, c_data, c_offsets, i);
+        }
+    }
+
+    static void constantStringVector(
+        const std::string & a,
+        const ColumnString::Chars_t & b_data,
+        const ColumnString::Offsets & b_offsets,
+        const TiDB::TiDBCollatorPtr & collator,
+        ColumnString::Chars_t & c_data,
+        ColumnString::Offsets & c_offsets)
+    {
+        StringOperationImpl::stringVectorConstant(b_data, b_offsets, a, collator, c_data, c_offsets);
+    }
+
+    static void constantConstant(
+        const std::string & a,
+        const std::string & b,
+        const TiDB::TiDBCollatorPtr & collator,
+        std::string & c)
+    {
+        LeastStringImpl::process(collator, a, b, c);
+    }
+};
+
+class FunctionLeastString : public IFunction
+{
+public:
+    static constexpr auto name = "tidbLeastString";
+    explicit FunctionLeastString() = default;
+    ;
+    static FunctionPtr create(const Context & context [[maybe_unused]])
+    {
+        return std::make_shared<FunctionLeastString>();
+    }
+
+    String getName() const override { return name; }
+    bool isVariadic() const override { return true; }
+
+    void setCollator(const TiDB::TiDBCollatorPtr & collator_) override
+    {
+        collator = collator_;
+    }
+
+    size_t getNumberOfArguments() const override { return 0; }
+    bool useDefaultImplementationForNulls() const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if (arguments.size() < 2)
+            throw Exception(
+                fmt::format("Number of arguments for function {} doesn't match: passed {}, should be at least 2.", getName(), arguments.size()),
+                ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+        for (const auto & argument : arguments)
+        {
+            if (!argument->isString())
+            {
+                throw Exception(
+                    fmt::format("argument type not string"),
+                    ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+            }
+        }
+        return std::make_shared<DataTypeString>();
+    }
+
+    void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result) const override
+    {
+        size_t num_arguments = arguments.size();
+        if (num_arguments < 2)
+        {
+            throw Exception(
+                fmt::format("Number of arguments for function {} doesn't match: passed {}, should be at least 2.", getName(), arguments.size()),
+                ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+        }
+        using impl = StringOperationImpl;
+        const auto * c0 = block.getByPosition(arguments[0]).column.get();
+        auto c_res = ColumnString::create();
+
+        for (size_t i = 1; i < num_arguments; ++i)
+        {
+            c_res = ColumnString::create();
```

Comment:
Instead of repeatedly creating result column, I think we can store the result directly in the result column.
Start: create result column and assign it to `res`
Step 1:  `c0` vs `c1`, store result in `res`
Step 2: `res` vs `c2`, more specifically, `res[0]` vs `c2[0]` store result in `res[0]`,  ... , `res[n]` vs `c2[n]` store result in `res[n]`. 
.....
Step n; `res` vs `cn`, store data in res.